File: front-end/src/types/domain/plan.ts
import type { WeekDay } from "../enums";

export type TrainingAppointmentDto = {
  day: number;
  time: string;
};

export type Plan = {
  id: number;
  title: string;
  description: string;
  trainerId: number;
  trainerName: string;
  sportId: number;
  sport: string;
  appointments: TrainingAppointmentDto[];
};

export type AddUpdatePlanDto = {
  title: string;
  description: string;
  trainerId: number;
  sportId: number;
  appointments: TrainingAppointmentDto[];
};

File: GMS-Bond/Model/Plan.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace GMS_Bond.Model
{
    public class Plan
    {
        public int Id { get; set; }
        public string Title { get; set; } = "";
        public string Description { get; set; } = "";

        [ForeignKey("Trainer")]
        public int TrainerId { get; set; }
        public virtual Trainer Trainer { get; set; } = null!; 

        [ForeignKey("Sport")]
        public int SportId { get; set; }
        public virtual Sport Sport { get; set; } = null!;

        public virtual ICollection<Session> Sessions { get; set; } = 
            new HashSet<Session>();
        public virtual ICollection<TrainingAppointment> TrainingAppointments { get; set; } = 
            new HashSet<TrainingAppointment>();
        public virtual ICollection<Package> Packages { get; set; } =
            new HashSet<Package>();

    }
}
